### PR TITLE
Add additional Squirrel parameters

### PR DESCRIPTION
**Pull request by pchinery**
Squirrel seems to have a few additional parameters, which I have added in this PR.

To have something to work with if things should change in future, I have also added "AdditionalArguments" to add raw arguments to the squirrel call. This is not a favorable solution to use, but rather a tool to create short-term workarounds. I'm open to discussing this part. It does help for the moment, but not having this option also has greatly accelerated me in creating the PR.

### CODE REVIEW

**Comment by pchinery** [comment]
With all due respect and understanding, I would like to amend my point above. I know that the FAKE team (an especially @matthid) does a terrific job at a very high pace and there are loots of completely reasons, why a PR is not merged or discussed. So I don't want to push attention to my PR when other things have a higher priority.

For these situations, it would be helpful to keep the workaround way in my PR of adding required parameters on a short, as this would have allowed me to continue with my work and would take the pressure from the team to merge it.

So again, please don't feel pushed in any way. I'm still open to discuss the "all purpose" parameter list of course, but my opinion of its usefulness has changed a bit.

**Comment by matthid** [comment]
Don't assume there are any other reasons than just missing time. Every contribution is appreciated. Sorry just give me a couple of days to review and give feedback

**Comment by pchinery** [comment]
Not a problem at all, missing time is one valid reasons for me, everyone has other work, family and time one might not want to spend on a computer. Waht I wanted to express: I fully understand any reason :)

**Comment by matthid** [comment]
Regarding the additional arguments field, afaik we have that on other modules as well. The only feedback is that is should be named similarly across modules

File: src/app/Fake.Installer.Squirrel/Squirrel.fs [hunk 1]
```diff
@@ -46,6 +52,12 @@ type ReleasifyParams = {
 
     /// The secret key for the code signing certificate
     SigningSecret : string option
+
+    /// Set the required .NET framework version, e.g. net461
+    FrameworkVersion : string option
+
+    /// Add other arguments, in case Squirrel adds new arguments in future
+    AdditionalArguments : string list
```
Comment (matthid) [inline src/app/Fake.Installer.Squirrel/Squirrel.fs]: It seems like we have any combination of this name already, besides the one you have chosen 💃 

Expecto:
```fsharp
      /// Custom arguments to use in the case the helper not yet supports them
      CustomArgs : string list
```

DotCover:
```fsharp
      CustomParameters: string
```

OpenCover:
```fsharp
          /// This options is used to add additional optional arguments, could be somthing like "-returntargetcode "
          OptionalArguments : string
```

InnoSetup
```fsharp
        /// Additional parameters
        AdditionalParameters : string option
```

Npm/Yarn
```fsharp
    | Custom of string
```

Chocolatey
```fsharp
        /// A character string containing additional arguments to give to choco.
        AdditionalArgs: string
```

I'm not sure but at least we should be consistent and I might enforce a name via contribution guidelines. Do you like any of the existing ones?

File: src/app/Fake.Installer.Squirrel/Squirrel.fs [hunk 2]
```diff
@@ -72,12 +88,19 @@ let private createSigningArgs (parameters : ReleasifyParams) =
     |> StringBuilder.toText
 
 let internal buildSquirrelArgs parameters nugetPackage =
-    new StringBuilder()
+    let sb = new StringBuilder()
+
+    parameters.AdditionalArguments |> Seq.iter (fun arg -> StringBuilder.appendIfNotNullOrEmpty arg "" sb |> ignore)
+
+    sb
     |> StringBuilder.appendIfNotNullOrEmpty nugetPackage "--releasify="
```
Comment (matthid) [inline src/app/Fake.Installer.Squirrel/Squirrel.fs]: I'd recommend updating this to the Arguments/CreateProcess API as the StringBuilder is a bit unsafe

File: src/app/Fake.Installer.Squirrel/Squirrel.fs (hunk 1 above)
Comment (matthid) [reply src/app/Fake.Installer.Squirrel/Squirrel.fs]: Also I'm not sure if it should be a list or just a string (and leave escaping to the user).

Comment (pchinery) [reply src/app/Fake.Installer.Squirrel/Squirrel.fs]: Facepalm for me... I'm sorry.

I think I would prefer AdditionalParameters. The way it is implemented right now, each parameter will be string escaped and surrounded with quotes. Not doing that is easy of course. In that case, I would change it to a string option. Do you have any preference between string option and string list? I can do the change on Sunday or Monday.

Comment (matthid) [reply src/app/Fake.Installer.Squirrel/Squirrel.fs]: I was more pointing out the bad quality of the existing code base than the name you have chosen. In fact I think one more name doesn't even matter at this point, but we should work in unifying this in the next version and I'd "lock" the one we choose now in the contribution guidelines :D

There are some tools where the "escaping" will actually break them so I tend to just provide a simple `string` here, but I'm open for arguments for using a list, which I guess would be more "safe".

File: src/app/Fake.Installer.Squirrel/Squirrel.fs (hunk 2 above)
Comment (pchinery) [reply src/app/Fake.Installer.Squirrel/Squirrel.fs]: I have now added a commit that uses the CreateProcess API and uses only a string option for the args

**Comment by matthid** [comment]
Thanks, looks good!

**Pull request merged**